src/lib/ggshield-api.ts

```diff
@@ -172,47 +172,42 @@ export async function scanFile(
     filePath,
   ]);
 
-  // Ignore errors concerning usage
-  // This occurs when the path of the file is invalid (i.e.VSCode sending an event for files not on the file system)
-  // or when the file is ignored in the .gitguardian.yaml
-  if (
-    proc.stderr.includes(
-      "Error: An ignored file or directory cannot be scanned"
-    )
-  ) {
-    updateStatusBarItem(StatusBarStatus.ignoredFile);
-    return;
-  }
-  if (proc.stderr.includes("Usage: ggshield secret scan path")) {
-    return undefined;
-  }
-  let errorMessage = "";
-  proc.stderr.split("\n").forEach((stderrLine) => {
+  if (proc.status === 128 || proc.status === 3) {
+    let errorMessage = "";
+    proc.stderr.split("\n").forEach((stderrLine) => {
+      if (
+        stderrLine.length > 0 &&
+        !stderrLine.includes("Scanning Path...") // ggshield outputs this info message on stderr, ignore it
+      ) {
+        errorMessage += stderrLine + "\n";
+      }
+    });
```

Comment:
I think using filter here is clearer :
```suggestion
    const errorMessage = proc.stderr.split("\n").filter(
        (stderrLine) => stderrLine.length > 0 && !stderrLine.includes("Scanning Path...")  // ggshield outputs this info message on stderr, ignore it
    ).join("\n");
```